src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractRemoveDocCommentNodeCodeFixProvider.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Roslyn.Utilities;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.DiagnosticComments.CodeFixes
+{
+    internal abstract class AbstractRemoveDocCommentNodeCodeFixProvider : CodeFixProvider
+    {
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public abstract override ImmutableArray<string> FixableDiagnosticIds { get; }
+        protected abstract string DocCommentSignifierToken { get; }
+
+        protected abstract SyntaxTriviaList GetRevisedDocCommentTrivia(string docCommentText);
+        protected abstract SyntaxNode GetDocCommentElementNode(SyntaxNode fullDocComentNode, TextSpan span);
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    FeaturesResources.Remove_tag,
```

Comment:
The reason I didn't use the `WellKnownFixAllProviders.BatchFixer` is because I didn't see it in use and thought you guys had originally decided not to use it internally, perhaps to squeeze the last millisecond out of performance.  If you wish, I could go through the code fixes and clean up all the various implementations of `MyCodeAction` to use `WellKnownFixAllProviders.BatchFixer`